pkg/controller/plan/adapter/vsphere/client.go

```diff
@@ -577,3 +577,58 @@ func (r *Client) DetachDisks(vmRef ref.Ref) (err error) {
 	// no-op
 	return
 }
+
+func (r *Client) GetNAAFromDatastore(ctx context.Context, datastoreRef ref.Ref) (string, error) {
```

Comment:
I don't see the need for an extra function, and an exported one. This could go into the getNAAFromDatastore as is.